### PR TITLE
ci: fix flaky cherrypy tests, use riot, snapshots

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
The cherrypy test cases were a little flaky due to the test server configuration that the cherrypy test case uses. 

This PR increases the wait time, removes the time assertions and introduces snapshot testing and ports the tests from tox to riot.

### CODE REVIEW

File: tests/contrib/cherrypy/test_middleware.py [hunk 1]
```diff
@@ -248,18 +208,14 @@ def test_unicode(self):
...
         assert_span_http_status_code(s, 200)
         assert s.meta.get(http.METHOD) == "GET"
         assert s.meta.get(http.URL) == u"http://127.0.0.1:54583/üŋïĉóđē"
 
     def test_404(self):
-        start = time.time()
         self.getPage(u"/404/test")
-        time.sleep(0.01)  # Without this here, span may not be ready for inspection, and timings can be incorrect.
-        end = time.time()
+        time.sleep(0.1)
```
Comment (jd) [inline tests/contrib/cherrypy/test_middleware.py:218]: do we still need those `time.sleep`? This is a really bad sign in general and proper to race condition. Is there any other way to make this working?

**Review by jd — DISMISSED** [review]

**Comment by jd** [comment]
one of the test still failed actually:
```
AssertionError: assert 0 == 1
 +  where 0 = len([])
self = <tests.contrib.cherrypy.test_middleware.TestCherrypy testMethod=test_unicode>

    def test_unicode(self):
        # Encoded utf8 query strings MUST be parsed correctly.
        # Here, the URL is encoded in utf8 and then %HEX
        # See https://docs.cherrypy.org/en/latest/_modules/cherrypy/test/test_encoding.html for more
        self.getPage(url_quote(u"/üŋïĉóđē".encode("utf-8")))
        self.assertStatus("200 OK")
        self.assertHeader("Content-Type", "text/html;charset=utf-8")
        self.assertBody(b"\xc3\xbc\xc5\x8b\xc3\xaf\xc4\x89\xc3\xb3\xc4\x91\xc4\x93")
    
        # ensure trace worked
        assert not self.tracer.current_span(), self.tracer.current_span().pprint()
        spans = self.tracer.writer.pop()
>       assert len(spans) == 1
E       AssertionError: assert 0 == 1
E        +  where 0 = len([])

tests/contrib/cherrypy/test_middleware.py:207: AssertionError
```

**Comment by Kyle-Verhoog** [comment]
yeah I'm going to take a stab at trying to make the requests synchronous as the current tests are always going to be flaky as-is

**Comment by Kyle-Verhoog** [comment]
Unfortunately there doesn't seem to be an easy way to achieve synchronicity here :confused:. CherryPy itself uses a sleep to test the hook that is used. 

https://github.com/cherrypy/cherrypy/blob/8c42f3145e6457969ecaac38fffbe65f1d1003e5/cherrypy/test/test_tools.py#L266-L273

I've made sure each test has the wait now.

File: tests/contrib/cherrypy/test_middleware.py (hunk 1 above)
Comment (Kyle-Verhoog) [reply tests/contrib/cherrypy/test_middleware.py:220]: Yeah it sucks. The way the CherryPy test helper works is by creating a subprocess to handle the requests. I tried to look for any kind of synchronization mechanism but there doesn't appear to be anything built into the helpers 😞

Comment (jd) [reply tests/contrib/cherrypy/test_middleware.py:220]: 😡

**Review by jd — APPROVED** [review]
would have been great to split this PR in multiple pieces though

**Pull request merged**